### PR TITLE
#223 Create Action Filter to Validate Methods of Controllers

### LINKED ISSUE #223: Create validation for Event and EventSchedule
-User should have permission
-Entity is not null

### PR DESCRIPTION
**Pull request by Anastasiagumenuk9**
Close #223

### CODE REVIEW

File: EventsExpress/ClientApp/src/services/EventsExpressService.js [hunk 1]
```diff
@@ -105,12 +105,16 @@ export default class EventsExpressService {
         return res;
     }
 
-    setEvent = async(data) => {
-        return this.setEventTemplate(data,'event/edit')
+    editEvent = async(data) => {
+        return this.setEventTemplate(data,`event/edit/?eventId=${data.id}`)
```
Comment (sand0) [inline EventsExpress/ClientApp/src/services/EventsExpressService.js]: use route bindings instead queryParams, psl

File: EventsExpress/Controllers/EventController.cs [hunk 2]
```diff
@@ -38,7 +38,8 @@ public class EventController : ControllerBase
         /// <response code="200">Create event proces success.</response>
         /// <response code="400">If Create process failed.</response>
         [HttpPost("[action]")]
-        public async Task<IActionResult> EditEventFromParent([FromForm] EventDto model)
+        [UserAccessTypeFilter]
+        public async Task<IActionResult> EditEventFromParent(Guid eventId, [FromForm] EventEditViewModel model)
```
Comment (sand0) [inline EventsExpress/Controllers/EventController.cs]: Move eventId to route

File: EventsExpress/ClientApp/src/services/EventsExpressService.js [hunk 3]
```diff
@@ -105,12 +105,16 @@ export default class EventsExpressService {
...
+    }
+
+    setEvent = async (data) => {
+        return this.setEventTemplate(data, `event/create`)
     }
 
     setEventFromParent = async (data) => {
-        return this.setEventTemplate(data,'event/EditEventFromParent');
+        return this.setEventTemplate(data,`event/EditEventFromParent/?eventId=${data.id}`);
     }
```
Comment (sand0) [inline EventsExpress/ClientApp/src/services/EventsExpressService.js]: Move events management logic into the separate file

**Review by sand0 — CHANGES_REQUESTED** [review]

File: EventsExpress/ClientApp/src/services/EventScheduleService.js [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+            file.append('Id', data.id);
+        }
+
+        file.append('Frequency', data.frequency);
+        file.append('LastRun', data.lastRun);
+        file.append('NextRun', data.nextRun);
+        file.append('Periodicity', data.periodicity);
+        file.append('IsActive', data.isActive);
+
+        const res = await baseService.setResourceWithData(`eventSchedule/${data.eventId}/edit`, file);
```
Comment (sand0) [inline EventsExpress/ClientApp/src/services/EventScheduleService.js:27]: shouldn't we use dataId here?

Comment (Anastasiagumenuk9) [reply EventsExpress/ClientApp/src/services/EventScheduleService.js:27]: Yes, it is required for Action Filter

File: EventsExpress/ClientApp/src/services/EventScheduleService.js [hunk 5]
```diff
@@ -0,0 +1,48 @@
...
+    }
+
+    getAllEventSchedules = async () => {
+        const res = await baseService.getResource(`eventSchedule/all`);
+        return res;
+    }
+
+    setEventSchedule = async (data) => {
+        let file = new FormData();
+        if (data.id != null) {
```
Comment (sand0) [inline EventsExpress/ClientApp/src/services/EventScheduleService.js]: Remove conditional here, we use this only for updating, so id is required

**Review by sand0 — COMMENTED** [review]
Most of the changes look good to me. 
@iuriikhrystiuk , could you check EventSceduleService.js and EventScheduleController 'cause I do not fully agree with the routing strategy but can't to propose constructive changes,

**Review by sand0 — APPROVED** [review]

**Pull request merged**